### ISSUE TITLE
"make install" installs headers and libraries into the build directory

### ISSUE THREAD

**Issue by realnc** [chrender/fizmo#21]
When running "make install", fizmo 0.8.4 installs files into the build directory. This is normally not an issue when building by hand, but when packaging this for distros, it's a headache.

For Gentoo, the build directory is:

/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build

And "make install" installs these:

```
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/lib/fizmo/libsndifsdl2.a
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/lib/fizmo/libmonospaceif.a
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/lib/fizmo/libfizmo.a
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/lib/fizmo/libdrilbo.a
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/lib/fizmo/libpixelif.a
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/lib/pkgconfig/libfizmo.pc
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/lib/pkgconfig/libmonospaceif.pc
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/lib/pkgconfig/libdrilbo.pc
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/lib/pkgconfig/libsndifsdl2.pc
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/lib/pkgconfig/libpixelif.pc
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/babel.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/blockbuf.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/blorb.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/cmd_hst.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/config.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/debugger.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/filelist.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/fizmo.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/history.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/hyphenation.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/iff.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/mathemat.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/misc.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/mt19937ar.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/object.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/output.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/property.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/routine.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/savegame.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/sound.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/stack.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/streams.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/table.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/text.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/undo.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/variable.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/wordwrap.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/zpu.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/interpreter/zscii.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/tools/filesys_c.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/tools/filesys.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/tools/i18n.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/tools/list.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/tools/stringmap.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/tools/tracelog.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/tools/types.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/tools/unused.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/tools/z_ucs.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/screen_interface/screen_interface.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/screen_interface/ScreenInterface.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/screen_interface/ScreenInterface.cpp
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/screen_interface/screen_monospace_interface.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/screen_interface/screen_pixel_interface.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/sound_interface/sound_interface.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/filesys_interface/filesys_interface.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/blorb_interface/blorb_interface.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/monospace_interface/monospace_interface.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/drilbo/drilbo.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/drilbo/drilbo-jpeg.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/drilbo/drilbo-mg1.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/drilbo/drilbo-png.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/drilbo/drilbo-ppm.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/drilbo/drilbo-x11.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/sound_sdl2/sound_sdl2.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/pixel_interface/pixel_interface.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/pixel_interface/true_type_factory.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/pixel_interface/true_type_font.h
/var/tmp/portage/games-engines/fizmo-0.8.4/work/fizmo-0.8.4/build/include/fizmo/pixel_interface/true_type_wordwrapper.h
```

**Comment by chrender** [comment]
You might try the following: Run `./configure` first, followed by a `make`, and then execute the following:

```
for libdir in libfizmo libmonospaceif libdrilbo libpixelif
do
  cd $libdir
  make install-data-local
  cd ..
done

for frontenddir in fizmo-console fizmo-ncursesw fizmo-sdl2
do
  cd $frontenddir
  make install
  cd ..
done
```

Fizmo is split into six libraries and five frontends. Since it's not much fun to have to download, build and install four to six libraries before one is able to build a frontend, the fizmo distribution package contains all  the libraries and development files along with the autofoo required to build all from a single invocation. Since the libraries and development files shouldn't be installed along with the frontends, but are required for the initial build, they're written to the “build” directory inside the distribution directory so the frontend build can find and use them.

Once `make` has been completed, the procedure above should cause the libraries  to only install the data files. The frontends don't install anything into the "build" subdir, so it's okay for them to execute normally.

**Comment by chrender** [comment]
Apologies – this was a complete misunderstanding on my side. I think I've fixed this in the current master branch(es) and created a beta version at https://fizmo.spellbreaker.org/source-beta/fizmo-0.8.5-b1.tar.gz for testing.

**Comment by chrender** [comment]
Fixed in fizmo release 0.8.5.
